tests/models/test_dag.py

```diff
@@ -677,87 +677,176 @@ def test_following_previous_schedule(self):
+        prev_utc = dag.previous_schedule(start_utc)
+        assert local_tz.convert(prev_utc).isoformat() == "2018-10-28T01:00:00+02:00"
+        prev_utc = dag.previous_schedule(prev_utc)
+        assert local_tz.convert(prev_utc).isoformat() == "2018-10-28T00:00:00+02:00"
 
-        prev = dag.previous_schedule(utc)
-        prev_local = local_tz.convert(prev)
+        # run forward a few
+        next_utc = dag.following_schedule(start_utc)
+        assert local_tz.convert(next_utc).isoformat() == "2018-10-28T02:00:00+01:00"
+        next_utc = dag.following_schedule(next_utc)
+        assert local_tz.convert(next_utc).isoformat() == "2018-10-28T03:00:00+01:00"
+        next_utc = dag.following_schedule(next_utc)
+        assert local_tz.convert(next_utc).isoformat() == "2018-10-28T04:00:00+01:00"
 
-        assert prev_local.isoformat() == "2018-10-26T03:00:00+02:00"
-        assert prev.isoformat() == "2018-10-26T01:00:00+00:00"
+    def test_following_previous_schedule_daily_dag_cest_to_cet_post(self):
+        # Second pass start on
+        local_tz = pendulum.timezone("Europe/Zurich")
+        start_local = local_tz.convert(datetime.datetime(2018, 10, 28, 3))
+
+        # start at 3:00+01, 1:00+00
+        assert start_local.isoformat() == "2018-10-28T03:00:00+01:00"
 
-        _next = dag.following_schedule(utc)
-        next_local = local_tz.convert(_next)
+        start_utc = timezone.convert_to_utc(start_local)
+        assert start_utc.isoformat() == "2018-10-28T02:00:00+00:00"
 
-        assert next_local.isoformat() == "2018-10-28T03:00:00+01:00"
-        assert _next.isoformat() == "2018-10-28T02:00:00+00:00"
+        dag = DAG("tz_dag", start_date=start_local, schedule="0 * * * *")
 
-        prev = dag.previous_schedule(_next)
-        prev_local = local_tz.convert(prev)
+        # run back a few
+        prev_utc = dag.previous_schedule(start_utc)
+        assert local_tz.convert(prev_utc).isoformat() == "2018-10-28T02:00:00+01:00"
+        prev_utc = dag.previous_schedule(prev_utc)
+        assert local_tz.convert(prev_utc).isoformat() == "2018-10-28T02:00:00+02:00"
+        prev_utc = dag.previous_schedule(prev_utc)
+        assert local_tz.convert(prev_utc).isoformat() == "2018-10-28T01:00:00+02:00"
 
-        assert prev_local.isoformat() == "2018-10-27T03:00:00+02:00"
-        assert prev.isoformat() == "2018-10-27T01:00:00+00:00"
+        # run forward a few
+        next_utc = dag.following_schedule(start_utc)
+        assert local_tz.convert(next_utc).isoformat() == "2018-10-28T04:00:00+01:00"
+        next_utc = dag.following_schedule(next_utc)
+        assert local_tz.convert(next_utc).isoformat() == "2018-10-28T05:00:00+01:00"
 
-    def test_following_previous_schedule_daily_dag_cet_to_cest(self):
+    def test_following_previous_schedule_daily_dag_cet_to_cest_pre(self):
         """
         Make sure DST transitions are properly observed
         """
+        # First pass, start on pre side
```

Comment:
starting here, I go through 2 sequences based on CET->CEST and CEST->CET. In both cases, I assure the behavior of getting next/previous from before and after the time change juncture. Please take care to review the behavior.